src/Go/boardAnalysis/goAI.ts

```diff
@@ -82,6 +86,13 @@ export function makeAIMove(boardState: BoardState, useOfflineCycles = true): Pro
   return Go.nextTurn;
 }
 
+export function resetAI(thinking = true) {
+  isAiThinking = thinking;
```

Comment:
Because `isAiThinking` is a boolean, it is prone to racy behavior. For instance, it will unconditionally be set to false in the finally block, even for a stale AI promise, leading to possible issues.

I think you're going to have to go back to storing a copy of `nextTurn` here instead of a boolean. That way, you can check if its value is still the same as what you started with, so you only clear it when it is correct to do so.

Reply:
I tweaked this to not update the semaphore if it was determined that the current AI process is pointed at a stale game state. which prevents orphaned processes from incorrectly updating that flag.

I tested a number of ways to set isAiThinking to be synched with nextTurn, but I ran into issues trying to keep it up-to-date when balancing the manual-play-vs-a-script flow with the script-vs-ai flow, and decided to identify the stale-game errors instead.